From 6f2aa9edc41621cf515778c2ac40e8f3e821a7b3 Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@qt.io>
Date: Thu, 2 Apr 2020 15:56:37 +0300
Subject: [PATCH] qt_record_extra_package_dependency: check that target exists

The target may not be defined which causes error:
get_target_property() called with non-existent target "qtwaylandscanner".

Change-Id: I58a9122456ccbbbb8fc9f0adce3b7ddcc985e6a4
---
 cmake/QtBuild.cmake | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/cmake/QtBuild.cmake b/cmake/QtBuild.cmake
index 80dc5f02fa..dfe88d7da4 100644
--- a/cmake/QtBuild.cmake
+++ b/cmake/QtBuild.cmake
@@ -2188,14 +2188,16 @@ endfunction()
 # main_target_name = qtwaylandscanner
 # dep_package_name = WaylandScanner
 function(qt_record_extra_package_dependency main_target_name dep_package_name dep_package_version)
-    get_target_property(extra_packages "${main_target_name}" QT_EXTRA_PACKAGE_DEPENDENCIES)
-    if(NOT extra_packages)
-        set(extra_packages "")
-    endif()
+    if (TARGET "${main_target_name}")
+        get_target_property(extra_packages "${main_target_name}" QT_EXTRA_PACKAGE_DEPENDENCIES)
+        if(NOT extra_packages)
+            set(extra_packages "")
+        endif()
 
-    list(APPEND extra_packages "${dep_package_name}\;${dep_package_version}")
-    set_target_properties("${main_target_name}" PROPERTIES QT_EXTRA_PACKAGE_DEPENDENCIES
-                                                           "${extra_packages}")
+        list(APPEND extra_packages "${dep_package_name}\;${dep_package_version}")
+        set_target_properties("${main_target_name}" PROPERTIES QT_EXTRA_PACKAGE_DEPENDENCIES
+                                                               "${extra_packages}")
+    endif()
 endfunction()
 
 # This function records a dependency between ${main_target_name} and ${dep_target_name}
-- 
2.17.1

